File: Python/20-perfect-number-or-not.py
num = int(input("Enter a number: "))
sum = 0

for i in range(1, num):
    if num%i == 0:
        sum += i

if sum == num:
    print(num, "is Perfect number.")
else:
    print(num, "is Not perfect number.")


# Using while loop

# i = 1
# while i < num:
#     if num%i == 0:
#         sum += i
#     i += 1

# if sum == num:
#     print(num, "is Perfect number.")
# else:
#     print(num, "is Not perfect number.")

File: Java/1-Important-Programs/20-perfect-number-or-not.java
import java.util.Scanner;

class Main{
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter a number:");
        int num = getInput.nextInt();
        int sum = 0;

        for (int i=1; i<num; i++){
            if (num%i == 0){
                sum += i;
            }
        }

        System.out.println((sum == num) ? (num + " is Perfect number.") : (num + " is Not perfect number."));
        getInput.close();
    }
}

// Using while loop
/* int i = 1;
        while (i<num) {
            if (num%i == 0){
                sum += i;
            }
            i++;
        } */